docs/DeveloperGuide.md

```diff
@@ -1,3 +1,1140 @@
+        > patients currently in the PatientList, which is an array of
+        > current Patients.
+
+4.  Printing the list of Patients
+
+    a.  If there are no patients currently in the list,
+        > PatientList.listPatients() calls
+        > PatientUI.emptyPatientListMessage(). And displays to the user
+        > that there are no patients in the list.
+
+    b.  If there are patients currently in the list,
+        > PatientList.listPatients() calls
+        > PatientUI.notEmptyPatientListMessage() before iterating
+        > through the list and calling PatientUI.printPatientList() to
+        > show each individual patient in the list.
+
+**Deleting a Patient:**
+
+delete \[Patient ID\]
+
+This feature allows the user to delete an existing patient in the list
+of patients. If the user fails to correctly input the proper parameters.
+Exceptions will be triggered that will notify the user on what caused
+the exception and the action will be aborted. The user will also be
+directed to use the help command to refer to the correct syntax.
+
+If the patient has been successfully deleted from the list they will be
+notified with a message showing the deleted patient's name and that they
+have been removed from the list.
+
+[Implementation:]{.ul}
+
+When a user attempts to add a patient the following classes will be
+accessed: PatientCommandInstance, UI, PatientUI, Patient, PatientList,
+PatientParser, PatientStorage.
+
+1.  User Executes command delete P12345
+
+    a.  PatientCommandInstance calls UI.scanInput() to read the user
+        > input
+
+    b.  PatientCommandInstance calls PatientParser.patientParse() to
+        > parse the user\'s input into a string array.
+
+2.  Checking user input in parser
+
+    a.  Using the case statements in patientParse, the command delete is
+        > identified.
+
+    b.  patientParse then calls lengthCheck() and iDParser() to scan the
+        > user input for any errors
+
+    c.  If lengthCheck() and iDParser() throw an exception, the current
+        > action will be aborted and the user will be notified with an
+        > appropriate error message and once again prompted to enter
+        > another command. If successful, the program will move to
+        > step 3.
+
+3.  Deleting the Patient from the list
+
+    a.  patientParse calls PatientList.deletePatient() and the patient
+        > specified by the user is deleted from the list through
+        > iterating through the patientList and discovering the location
+        > of Patient specified before deleting the object from the
+        > ArrayList
+
+4.  Notifying user that Patient has been deleted
+
+    a.  After successfully deleting the Patient from the PatientList
+        > object. PatientList.deletePatient() calls
+        > PatientUI.deletePatientMessage() to print a message notifying
+        > the user that the Patient has been deleted.
+
+5.  Saving the current list of patients
```

Comment:
These steps are very well detailed, but maybe consider trimming them to only include the important/essential parts? Considering that the method being explained is 'deleting a patient', maybe more focus could be drawn to the actual Patient and related classes rather than the Parser functions? Maybe there could be some reorganization so that Parser has a section explaining its implementation separately

![image](https://user-images.githubusercontent.com/57757704/113083233-6b295380-920e-11eb-8608-53359b3f4712.png)